core/src/lib.rs

```diff
@@ -521,6 +550,20 @@ pub struct VerticalExtremes {
 }
 
 impl VerticalExtremes {
+    /// Create a new `VerticalExtremes` from the two provided values.
+    ///
+    /// Panics if either value is `NaN`.
+    #[inline]
+    #[must_use]
+    pub fn new(lowest: f64, highest: f64) -> Self {
```

Comment:
It's a shame that we lose the explicit names when instantiating, but I get the trade-off